gameShop/src/main/java/org/example/service/Singleton.java

```diff
@@ -0,0 +1,39 @@
+package org.example.service;
+
+import lombok.Getter;
+import org.example.AppData;
+import org.example.repository.GameRepositoryImpl;
+import org.example.repository.UserRepositoryImpl;
+import org.example.repository.dao.GameRepository;
+import org.example.repository.dao.UserRepository;
+
+import java.sql.Connection;
+
+@Getter
+public class Singleton {
+
+    private static Singleton instance;
+    private final UserService userService;
+    private final GameService gameService;
+    private final UserDataService userDataService;
+    private final AppData appData;
+
+    private Singleton(Connection connection) {
+        UserRepository userRepository = new UserRepositoryImpl(connection);
+        userService = new UserService(userRepository);
+
+        GameRepository gameRepository = new GameRepositoryImpl(connection);
+        gameService = new GameService(gameRepository);
+
+        userDataService = new UserDataService(userRepository, gameRepository);
+
+        appData = AppData.getInstance();
+    }
+
+    public static Singleton getInstance(Connection connection) {
+        if (instance == null) {
+            instance = new Singleton(connection);
+        }
+        return instance;
+    }
```

Comment:
this part is a potential problem, what if I use this method twice with 2 different connections to different databases? Class needs to be rewritten and simplified